xarray/backends/netCDF4_.py

```diff
@@ -317,6 +317,39 @@ def _is_list_of_strings(value) -> bool:
     return arr.dtype.kind in ["U", "S"] and arr.size > 1
 
 
+def _build_and_get_enum(
+    store, var_name: str, dtype: np.dtype, enum_name: str, enum_dict: dict[str, int]
+) -> Any:
+    """
+    Add or get the netCDF4 Enum based on the dtype in encoding.
+    The return type should be ``netCDF4.EnumType``,
```

Comment:
Maybe this is contrary to usual xarray practices, but could you use the `TYPE_CHECKING` import guard to get the return type?

Reply:
I'm (still) not comfortable with all that typing stuff. If you could make a suggestion, that would be really helpful.